### ISSUE TITLE
onRowReorder emits same event when dropping directly above or two above the dragged row

### ISSUE THREAD

**Issue by LucaVazz** [primefaces/primeng#11067]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
https://stackblitz.com/edit/primeng-tablereorder-demo-wg5ap7

**Current behavior**
When using drag and drop to re-order a row, the event emitted to onRowReorder is the same when moving the item to the line on top or two up from the dragged row.

I.e. when reordering and dropping the row marked in red to the line marked in light blue or the line marked in dark blue, both produce `{ dragIndex: 3, dropIndex: 2}` as the event.
Dropping it at the line marked in green produces `{ dragIndex: 3, dropIndex: 3}` as the event.

<img src="https://user-images.githubusercontent.com/4174525/149324848-012121a0-d30d-467a-a453-0872eb286251.png" width="350px">

**Expected behavior**
Dropping the element at the lines marked in light blue or dropping it at the line marked in dark blue should produce a different event.

Perhaps dropping it at the line marked in light blue should be an empty event, since no reorder happens in that case, or emit  `{ dragIndex: 3, dropIndex: 3}` as the event same as when dropping it at the line marked in green.

**What is the motivation / use case for changing the behavior?**
n/a

**Please tell us about your environment:**
n/a

* **Angular version:** 14.2

* **PrimeNG version:** 14.2

* **Browser:** all
 
* **Language:** all

* **Node (for AoT issues):** n/a

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you reopen the issue so we can include it in our roadmap? 
Please don't forget to add your feedback as a comment after reopening the issue. These will be taken into account by us and will contribute to the development of this feature. Thanks a lot for your understanding!

Best Regards,

**Comment by LucaVazz** [comment]
@mertsincan Unfortunately this issue is still present with angular + primeng 14.2 - I've updated the stackblitz example

Could you please reopen the issue so it can be investigated by the team?

**Comment by Superhands89** [comment]
1 up. Waiting on a fix for this.

**Comment by NateRadebaugh** [comment]
@mertsincan can you please take another look at this? This is a pretty glaring issue that makes the reorder functionality pretty much non-viable on a production product.

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
